# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/nsHttpChannel.cpp
# Commit: f7b6aac1b58b
# Full Hash: f7b6aac1b58bc959e431136e1b23231abe55d0b1
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2020-07-16 21:27:37
# Regressor Bug: 1497249
# File Overlap Count: 4
# Description:
#   Bug 1497249 - P2: ipdl for nsIWebSocketConnection r=michal
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D30624
# ==============================================================================

diff -r 74bb3c2ab767 -r f7b6aac1b58b netwerk/protocol/http/nsHttpChannel.cpp
--- a/netwerk/protocol/http/nsHttpChannel.cpp	Thu Jul 16 12:59:23 2020 +0000
+++ b/netwerk/protocol/http/nsHttpChannel.cpp	Thu Jul 16 13:34:58 2020 +0000
@@ -7942,6 +7942,13 @@
                         mResponseHead->Status() == 200;
 
   if (upgradeWebsocket || upgradeConnect) {
+    // TODO: Support connection upgrade for socket process in bug 1632809.
+    if (nsIOService::UseSocketProcess() && upgradeConnect) {
+      Unused << mUpgradeProtocolCallback->OnUpgradeFailed(
+          NS_ERROR_NOT_IMPLEMENTED);
+      return ContinueOnStopRequest(aStatus, aIsFromNet, aContentComplete);
+    }
+
     nsresult rv = gHttpHandler->CompleteUpgrade(aTransWithStickyConn,
                                                 mUpgradeProtocolCallback);
     if (NS_FAILED(rv)) {